internal/backends/winit/winitwindowadapter.rs

```diff
@@ -653,32 +653,10 @@ impl WinitWindowAdapter {
         window_inner
             .context()
             .spawn_local(async move {
-                let Ok(settings) = ashpd::desktop::settings::Settings::new().await else { return };
-
-                let Ok(initial_color_scheme_value) = settings.color_scheme().await else { return };
-
-                let convert = |ashpd_color_scheme| match ashpd_color_scheme {
-                    ashpd::desktop::settings::ColorScheme::NoPreference => ColorScheme::Unknown,
-                    ashpd::desktop::settings::ColorScheme::PreferDark => ColorScheme::Dark,
-                    ashpd::desktop::settings::ColorScheme::PreferLight => ColorScheme::Light,
-                };
-
-                if let Some(window) = self_weak.upgrade() {
-                    window.set_color_scheme(convert(initial_color_scheme_value));
-                }
-
-                let Ok(mut color_scheme_stream) = settings.receive_color_scheme_changed().await
-                else {
-                    return;
-                };
-
-                loop {
-                    use futures::stream::StreamExt;
-
-                    let Some(new_scheme) = color_scheme_stream.next().await else { break };
-                    if let Some(window) = self_weak.upgrade() {
-                        window.set_color_scheme(convert(new_scheme));
-                    }
+                if let Err(err) =
+                    crate::xdg_color_scheme::XdgColorSchemeWatcher::watch(self_weak).await
+                {
+                    i_slint_core::debug_log!("Error watching for xdg color schemes: {}", err);
```

Comment:
I'm wondering in which cases we get this error. 
For example, when not running on a desktop that has a org.freedesktop.portal.Settings?
In that case, should we not silently ignore the error?
But I don't know if that's an error or not.

Reply:
I wasn't sure either. The errors I've seen so far were due to bugs on my side (I used wrong types in dbus signature, for example). So in the absence of that I opted for this, but I'm open to other suggestions.